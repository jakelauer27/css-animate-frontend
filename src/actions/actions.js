export const signIn = (user) => ({
  type: 'SIGN_IN',
  user
})

export const signOut = () => ({
  type: 'SIGN_OUT'
})

export const loadPrebuiltAnimations = (prebuiltAnimations) => ({
  type: 'LOAD_PREBUILT_ANIMATIONS',
  prebuiltAnimations
})

export const updateCurrentAnimation = (animation) => ({
  type: 'UPDATE_CURRENT_ANIMATION',
  animation
})

export const saveOriginalAnimation = (animation) => ({
  type: 'SAVE_ORIGINAL_ANIMATION',
  animation
})

export const loadMyAnimations = (myAnimations) => ({
  type: 'LOAD_MY_ANIMATIONS',
  myAnimations
})

export const deleteMyAnimation = (id) => ({
  type: 'DELETE_MY_ANIMATION',
  id
})

export const selectAnimationForEdit = (animation) => ({
  type: 'SELECT_ANIMATION_FOR_EDIT',
  animation
})

export const removeAnimationForEdit = () => ({
  type: 'REMOVE_ANIMATION_FOR_EDIT'
})


