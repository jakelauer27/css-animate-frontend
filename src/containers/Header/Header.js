import React, { Component } from 'react'
import { Link, withRouter } from 'react-router-dom'
import { connect } from 'react-redux'
import { PropTypes } from 'prop-types'
import { signOut } from '../../actions/actions'

export class Header extends Component {

  render() {
    const { user, signOut } = this.props
    let greeting = ''
    let login = 'Login'
    if (user.name) {
      login = 'Logout'
      greeting = `Hi ${user.name}!`
    }

    return (
      <header>
        <h1 className='main-title'>CSS ani<span>Mate</span></h1>
        <div className='header-btn-container'>
        <Link to={`properties/selectAnimation`}>
          <button className='load-example-btn'>Load Animation</button>
        </Link>
        </div>
        <div className='inst-log-container'>
          <h3 className='user-greeting'>{greeting}</h3>
          <Link to={`properties/howto/general`}>
            <button className='questions-btn'>Instructions</button>
          </Link>
          <Link to='properties/login'>
            <button className='login-btn' onClick={() => signOut()}>{login}</button>
          </Link>
        </div>
      </header>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export const mapDispatchToProps = (dispatch) => ({
  signOut: () => dispatch(signOut())
})

Header.propTypes = {
  user: PropTypes.object,
  signOut: PropTypes.func.isRequired
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Header))