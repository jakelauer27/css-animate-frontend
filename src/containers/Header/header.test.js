import { Header, mapStateToProps, mapDispatchToProps } from './Header'
import React from 'react'
import { shallow } from 'enzyme'
import { signOut } from '../../actions/actions'

describe('header', () => {
  let mockSignOut
  let wrapper

  beforeEach(() => {
    mockSignOut = jest.fn()
    wrapper = shallow(<Header signOut={mockSignOut} user={{name: null}} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot if a user is logged in', () => {
    wrapper = shallow(<Header signOut={mockSignOut} user={{name: 'jake'}} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should called signOut when the login/logout button is clicked', () => {
    wrapper.find('.login-btn').simulate('click')
    expect(mockSignOut).toHaveBeenCalled()
  })

  describe('mapStateToProps', () => {
        
    it('should return a props object with a user property', () => {
      const mockState = {user: {name: 'jake', id: 1}}
      const expected = {user: {name: 'jake', id: 1}}

      expect(mapStateToProps(mockState)).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {

    it('calls dispatch with a signIn action when loginUser is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = signOut()
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.signOut()
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})