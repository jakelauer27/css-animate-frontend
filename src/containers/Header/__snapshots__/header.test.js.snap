// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    signOut={[MockFunction]}
    user={
      Object {
        "name": null,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="main-title"
        >
          CSS ani
          <span>
            Mate
          </span>
        </h1>,
        <div
          className="header-btn-container"
        >
          <Link
            replace={false}
            to="properties/selectAnimation"
          >
            <button
              className="load-example-btn"
            >
              Load Animation
            </button>
          </Link>
        </div>,
        <div
          className="inst-log-container"
        >
          <h3
            className="user-greeting"
          >
            
          </h3>
          <Link
            replace={false}
            to="properties/howto/general"
          >
            <button
              className="questions-btn"
            >
              Instructions
            </button>
          </Link>
          <Link
            replace={false}
            to="properties/login"
          >
            <button
              className="login-btn"
              onClick={[Function]}
            >
              Login
            </button>
          </Link>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "CSS ani",
            <span>
              Mate
            </span>,
          ],
          "className": "main-title",
        },
        "ref": null,
        "rendered": Array [
          "CSS ani",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Mate",
            },
            "ref": null,
            "rendered": "Mate",
            "type": "span",
          },
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="properties/selectAnimation"
          >
            <button
              className="load-example-btn"
            >
              Load Animation
            </button>
          </Link>,
          "className": "header-btn-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="load-example-btn"
            >
              Load Animation
            </button>,
            "replace": false,
            "to": "properties/selectAnimation",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Load Animation",
              "className": "load-example-btn",
            },
            "ref": null,
            "rendered": "Load Animation",
            "type": "button",
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="user-greeting"
            >
              
            </h3>,
            <Link
              replace={false}
              to="properties/howto/general"
            >
              <button
                className="questions-btn"
              >
                Instructions
              </button>
            </Link>,
            <Link
              replace={false}
              to="properties/login"
            >
              <button
                className="login-btn"
                onClick={[Function]}
              >
                Login
              </button>
            </Link>,
          ],
          "className": "inst-log-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "user-greeting",
            },
            "ref": null,
            "rendered": "",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="questions-btn"
              >
                Instructions
              </button>,
              "replace": false,
              "to": "properties/howto/general",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Instructions",
                "className": "questions-btn",
              },
              "ref": null,
              "rendered": "Instructions",
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="login-btn"
                onClick={[Function]}
              >
                Login
              </button>,
              "replace": false,
              "to": "properties/login",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
                "className": "login-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Login",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="main-title"
          >
            CSS ani
            <span>
              Mate
            </span>
          </h1>,
          <div
            className="header-btn-container"
          >
            <Link
              replace={false}
              to="properties/selectAnimation"
            >
              <button
                className="load-example-btn"
              >
                Load Animation
              </button>
            </Link>
          </div>,
          <div
            className="inst-log-container"
          >
            <h3
              className="user-greeting"
            >
              
            </h3>
            <Link
              replace={false}
              to="properties/howto/general"
            >
              <button
                className="questions-btn"
              >
                Instructions
              </button>
            </Link>
            <Link
              replace={false}
              to="properties/login"
            >
              <button
                className="login-btn"
                onClick={[Function]}
              >
                Login
              </button>
            </Link>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "CSS ani",
              <span>
                Mate
              </span>,
            ],
            "className": "main-title",
          },
          "ref": null,
          "rendered": Array [
            "CSS ani",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Mate",
              },
              "ref": null,
              "rendered": "Mate",
              "type": "span",
            },
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="properties/selectAnimation"
            >
              <button
                className="load-example-btn"
              >
                Load Animation
              </button>
            </Link>,
            "className": "header-btn-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="load-example-btn"
              >
                Load Animation
              </button>,
              "replace": false,
              "to": "properties/selectAnimation",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Load Animation",
                "className": "load-example-btn",
              },
              "ref": null,
              "rendered": "Load Animation",
              "type": "button",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="user-greeting"
              >
                
              </h3>,
              <Link
                replace={false}
                to="properties/howto/general"
              >
                <button
                  className="questions-btn"
                >
                  Instructions
                </button>
              </Link>,
              <Link
                replace={false}
                to="properties/login"
              >
                <button
                  className="login-btn"
                  onClick={[Function]}
                >
                  Login
                </button>
              </Link>,
            ],
            "className": "inst-log-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "user-greeting",
              },
              "ref": null,
              "rendered": "",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="questions-btn"
                >
                  Instructions
                </button>,
                "replace": false,
                "to": "properties/howto/general",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Instructions",
                  "className": "questions-btn",
                },
                "ref": null,
                "rendered": "Instructions",
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="login-btn"
                  onClick={[Function]}
                >
                  Login
                </button>,
                "replace": false,
                "to": "properties/login",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login",
                  "className": "login-btn",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Login",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`header should match the snapshot if a user is logged in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    signOut={[MockFunction]}
    user={
      Object {
        "name": "jake",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="main-title"
        >
          CSS ani
          <span>
            Mate
          </span>
        </h1>,
        <div
          className="header-btn-container"
        >
          <Link
            replace={false}
            to="properties/selectAnimation"
          >
            <button
              className="load-example-btn"
            >
              Load Animation
            </button>
          </Link>
        </div>,
        <div
          className="inst-log-container"
        >
          <h3
            className="user-greeting"
          >
            Hi jake!
          </h3>
          <Link
            replace={false}
            to="properties/howto/general"
          >
            <button
              className="questions-btn"
            >
              Instructions
            </button>
          </Link>
          <Link
            replace={false}
            to="properties/login"
          >
            <button
              className="login-btn"
              onClick={[Function]}
            >
              Logout
            </button>
          </Link>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "CSS ani",
            <span>
              Mate
            </span>,
          ],
          "className": "main-title",
        },
        "ref": null,
        "rendered": Array [
          "CSS ani",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Mate",
            },
            "ref": null,
            "rendered": "Mate",
            "type": "span",
          },
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="properties/selectAnimation"
          >
            <button
              className="load-example-btn"
            >
              Load Animation
            </button>
          </Link>,
          "className": "header-btn-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="load-example-btn"
            >
              Load Animation
            </button>,
            "replace": false,
            "to": "properties/selectAnimation",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Load Animation",
              "className": "load-example-btn",
            },
            "ref": null,
            "rendered": "Load Animation",
            "type": "button",
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="user-greeting"
            >
              Hi jake!
            </h3>,
            <Link
              replace={false}
              to="properties/howto/general"
            >
              <button
                className="questions-btn"
              >
                Instructions
              </button>
            </Link>,
            <Link
              replace={false}
              to="properties/login"
            >
              <button
                className="login-btn"
                onClick={[Function]}
              >
                Logout
              </button>
            </Link>,
          ],
          "className": "inst-log-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hi jake!",
              "className": "user-greeting",
            },
            "ref": null,
            "rendered": "Hi jake!",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="questions-btn"
              >
                Instructions
              </button>,
              "replace": false,
              "to": "properties/howto/general",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Instructions",
                "className": "questions-btn",
              },
              "ref": null,
              "rendered": "Instructions",
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="login-btn"
                onClick={[Function]}
              >
                Logout
              </button>,
              "replace": false,
              "to": "properties/login",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Logout",
                "className": "login-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Logout",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="main-title"
          >
            CSS ani
            <span>
              Mate
            </span>
          </h1>,
          <div
            className="header-btn-container"
          >
            <Link
              replace={false}
              to="properties/selectAnimation"
            >
              <button
                className="load-example-btn"
              >
                Load Animation
              </button>
            </Link>
          </div>,
          <div
            className="inst-log-container"
          >
            <h3
              className="user-greeting"
            >
              Hi jake!
            </h3>
            <Link
              replace={false}
              to="properties/howto/general"
            >
              <button
                className="questions-btn"
              >
                Instructions
              </button>
            </Link>
            <Link
              replace={false}
              to="properties/login"
            >
              <button
                className="login-btn"
                onClick={[Function]}
              >
                Logout
              </button>
            </Link>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "CSS ani",
              <span>
                Mate
              </span>,
            ],
            "className": "main-title",
          },
          "ref": null,
          "rendered": Array [
            "CSS ani",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Mate",
              },
              "ref": null,
              "rendered": "Mate",
              "type": "span",
            },
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="properties/selectAnimation"
            >
              <button
                className="load-example-btn"
              >
                Load Animation
              </button>
            </Link>,
            "className": "header-btn-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="load-example-btn"
              >
                Load Animation
              </button>,
              "replace": false,
              "to": "properties/selectAnimation",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Load Animation",
                "className": "load-example-btn",
              },
              "ref": null,
              "rendered": "Load Animation",
              "type": "button",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="user-greeting"
              >
                Hi jake!
              </h3>,
              <Link
                replace={false}
                to="properties/howto/general"
              >
                <button
                  className="questions-btn"
                >
                  Instructions
                </button>
              </Link>,
              <Link
                replace={false}
                to="properties/login"
              >
                <button
                  className="login-btn"
                  onClick={[Function]}
                >
                  Logout
                </button>
              </Link>,
            ],
            "className": "inst-log-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hi jake!",
                "className": "user-greeting",
              },
              "ref": null,
              "rendered": "Hi jake!",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="questions-btn"
                >
                  Instructions
                </button>,
                "replace": false,
                "to": "properties/howto/general",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Instructions",
                  "className": "questions-btn",
                },
                "ref": null,
                "rendered": "Instructions",
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="login-btn"
                  onClick={[Function]}
                >
                  Logout
                </button>,
                "replace": false,
                "to": "properties/login",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Logout",
                  "className": "login-btn",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Logout",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
